### DISCUSSION TITLE
blog/how-to-run-background-jobs-on-vercel-without-a-queue

### DISCUSSION THREAD

**Discussion by giscus** [zackproser/portfolio#445] [Announcements]
# blog/how-to-run-background-jobs-on-vercel-without-a-queue

I recently needed to figure out how to have my Vercel API route accept a new job and return a response immediately while still performing long running processing in the background. This is what I learned. 

https://zackproser.com/blog/how-to-run-background-jobs-on-vercel-without-a-queue

**Comment by prageeth** [comment]
How do you handle retries if a background request fails? I faced similar issues, which led me to build a proper queue with a simple HTTP API for dispatching. Would love to know what you think:
https://dispatched.dev/

**Comment by ralphsmith80** [comment]
I used this recently, but I don't know if it's necessary. How would this differ from using `after` (fire and forget) or `waitUntil` (don't block the response, but wait until completion before exiting the function execution)? 

https://vercel.com/docs/functions/vercel-functions-package#using-after-in-nextjs
https://vercel.com/docs/functions/vercel-functions-package#waituntil

**Reply by prageeth** [reply]
Agreed, for the basic cases you can use `after()`, but it doesn't handle all cases where queues are generally used. That's why I developed [`dispatched.dev`](https://dispatched.dev/). It doesn't simply run something in the background, it has other capabilities like scheduling and retries which are crucial when talking to third party integrations and failures can happen.
